Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/distance.h

```diff
@@ -1310,8 +1322,1295 @@ double approximate_symmetric_Hausdorff_distance(const TriangleMesh& tm1,
+  // std::cout << "* num cores: " << nb_cores << std::endl;
+
+  if (
+    boost::is_convertible<Concurrency_tag, CGAL::Parallel_tag>::value &&
+    nb_cores > 1 && faces(tm1).size() >= min_nb_faces_to_split) {
+
+    // (0) -- Compute infinity value.
+    timer.reset();
+    timer.start();
+    const auto bbox1 = bbox(tm1);
+    const auto bbox2 = bbox(tm2);
+    const auto bb = bbox1 + bbox2;
+    const FT sq_dist = CGAL::squared_distance(
+      Point_3(bb.xmin(), bb.ymin(), bb.zmin()),
+      Point_3(bb.xmax(), bb.ymax(), bb.zmax()));
+    infinity_value = CGAL::approximate_sqrt(sq_dist) * FT(2);
+    CGAL_assertion(infinity_value >= FT(0));
+    timer.stop();
+    // const double time0 = timer.time();
+    // std::cout << "- computing infinity (sec.): " << time0 << std::endl;
+
+    // (1) -- Create partition of tm1.
+    timer.reset();
+    timer.start();
+    using Face_property_tag = CGAL::dynamic_face_property_t<int>;
+    auto face_pid_map = get(Face_property_tag(), tm1);
+    CGAL::METIS::partition_graph(
+      tm1, nb_cores, CGAL::parameters::
+      face_partition_id_map(face_pid_map));
+    timer.stop();
+    // const double time1 = timer.time();
+    // std::cout << "- computing partition time (sec.): " << time1 << std::endl;
+
+    // (2) -- Create a filtered face graph for each part.
+    timer.reset();
+    timer.start();
+    tm1_parts.reserve(nb_cores);
+    for (int i = 0; i < nb_cores; ++i) {
+      tm1_parts.emplace_back(tm1, i, face_pid_map);
+      // CGAL_assertion(tm1_parts.back().is_selection_valid()); // TODO: why is it triggered sometimes?
+      // std::cout << "- part " << i << " size: " << tm1_parts.back().number_of_faces() << std::endl;
+    }
+    CGAL_assertion(tm1_parts.size() == nb_cores);
+    timer.stop();
+    // const double time2 = timer.time();
+    // std::cout << "- creating graphs time (sec.): " << time2 << std::endl;
+
+    // (3) -- Preprocess all input data.
+    timer.reset();
+    timer.start();
+    tm1_trees.resize(tm1_parts.size());
+    tm_wrappers.reserve(tm1_parts.size() + 1);
+    for (std::size_t i = 0; i < tm1_parts.size(); ++i) {
+      tm_wrappers.push_back(TM1_wrapper(tm1_parts[i], vpm1, false, tm1_trees[i]));
+    }
+    tm_wrappers.push_back(TM2_wrapper(tm2, vpm2, true, tm2_tree));
+    CGAL_assertion(tm_wrappers.size() == tm1_parts.size() + 1);
+    Bounded_error_preprocessing<TM1_wrapper, TM2_wrapper> bep(tm_wrappers);
+    tbb::parallel_reduce(tbb::blocked_range<std::size_t>(0, tm_wrappers.size()), bep);
+    timer.stop();
+    // const double time3 = timer.time();
+    // std::cout << "- creating trees time (sec.) " << time3 << std::endl;
+
+    // Final timing.
+    // std::cout << "* preprocessing parallel time (sec.) " <<
+    //   time0 + time1 + time2 + time3 << std::endl;
+
+  } else // sequential version
+  #endif // defined(CGAL_LINKED_WITH_TBB) && defined(CGAL_METIS_ENABLED)
+  {
+    // std::cout << "* preprocessing sequential version " << std::endl;
```

Comment:
I suggest to use a macro to hide debug code rather than having it commented.

Reply:
Done.